internal/distributed/querynode/service.go

```diff
@@ -241,19 +241,22 @@ func (s *Server) Stop() (err error) {
 		logger.Info("QueryNode stopped", zap.Error(err))
 	}()
 
+	logger.Info("internal server[querynode] start to stop")
 	err = s.querynode.Stop()
 	if err != nil {
+		log.Error("failed to close querynode", zap.Error(err))
 		return err
 	}
 	if s.etcdCli != nil {
 		defer s.etcdCli.Close()
 	}
 
-	s.cancel()
 	if s.grpcServer != nil {
 		utils.GracefulStopGRPCServer(s.grpcServer)
 	}
 	s.wg.Wait()
+
+	s.cancel()
```

Comment:
should be ahead of s.wg.Wait()?

Reply:
wg.Wait() is only used to wait grpc server to exit